File: ICodeGuru/349-IntersectionOfTwoArrays/11-ContainerWithMostWater/ContainerWithMostWater.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        max_area = 0
        left = 0
        right = len(height) - 1
        while left <right:
            cur_area = min(height[left], height[right]) * (right - left)
            max_area = max(cur_area , max_area)
            if(height[left] < height[right]):
                left += 1
            else:
                right -= 1
        return max_area

File: Y_cpp/11-ContainerWithMostWater/ContainerWithMostWater.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        //Optimized Approach O(n) --> Time Complexity
        int lp = 0 , ans = 0;
        int rp = height.size() - 1;
        while(lp < rp){
            int width = rp - lp;
            int ht = min(height[lp],height[rp]);
            int area = width * ht;
            
            ans = max(area,ans);

            height[lp] < height[rp] ? lp++ : rp--;
        }
        return ans;








        //Brute Force Approach O(n^2) --> Time Complexity (TLE)

        // int ans = 0;
        // for(int i = 0; i < height.size(); i++){
        //     for(int j = i+1; j < height.size(); j++){
        //         int width = j - i;
        //         int ht = min(height[i],height[j]);
        //         int area = width * ht;
        //         ans = max(ans, area);
        //     }
        // }
        // return ans;
    }
};